clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/AbstractAccountCredentials.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.security;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.netflix.spinnaker.fiat.model.Authorization;
+import com.netflix.spinnaker.fiat.model.resources.Permissions;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public abstract class AbstractAccountCredentials<T> implements AccountCredentials<T> {
```

Comment:
This seems wrong to me. `getCredentials` and `getPermissions` both are used fairly widely in the implementations, but is no longer part of the `AccountCredentials` API - we're basically forcing people to extend from this class, which is quite limiting.

I'm thinking it might make more sense to move the `fiat-api` models into an actual shared models location that we can feel comfortable having as a dependency. I'm happy to bite this work off in support of this PR.

Reply:
`getCredentials` is in AccountCredentials, but also here to separate the jackson annotation. Something that I was going to do with mixins later.
getPermissions can't be in `-api` because of the need to include fiat. I agree with a change to fiat-api in the long term. I was just trying to find a palatable way to move stuff to `-api` without dependencies thinking that we could make those changes later.
This could be an interface with default methods rather than an abstract class if the need to extend it is a concern.

Reply:
I see. I'll look into taking on something WRT Fiat, since this appears to be a reoccurring issue in the plugin effort. This is good for now - can you add a TODO to this class so this isn't lost?